lib/SPIRV/libSPIRV/SPIRVInstruction.cpp

```diff
@@ -244,8 +244,14 @@ SPIRVInstruction *createInstFromSpecConstantOp(SPIRVSpecConstantOp *Inst) {
   assert(isSpecConstantOpAllowedOp(OC) &&
          "Op code not allowed for OpSpecConstantOp");
   Ops.erase(Ops.begin(), Ops.begin() + 1);
-  return SPIRVInstTemplateBase::create(OC, Inst->getType(), Inst->getId(), Ops,
-                                       nullptr, Inst->getModule());
+  switch (OC) {
```

Comment:
I'd like to avoid one more opcode-based switch. But to make it work we should change `SPIRVSelect` to inherit it from `SPIRVInstTemplateBase`, which is probably out of scope for this PR. Other than that, LGTM.